amethyst_assets/src/asset.rs

```diff
@@ -33,10 +33,64 @@ pub trait Format<A: Asset>: Send + 'static {
     type Options: Send + 'static;
 
     /// Reads the given bytes and produces asset data.
+    ///
+    /// ## Reload
+    ///
+    /// The reload structure has metadata which allows the asset management
+    /// to reload assets if necessary (for hot reloading).
+    /// You should only create this if `create_reload` is `true`.
+    /// Also, the parameter is just a request, which means you can also return `None`.
     fn import(
         &self,
         name: String,
```

Comment:
`name` and `options` should be passed by reference to avoid unnecessary cloning.

Reply:
And in `SimpleFormat` as well. Otherwise cloning will be necessary 😄

Reply:
I'm aware of the fact that an allocation is not desirable, but we can't point to the stack in a multi-threaded system.

Reply:
I don't follow. If you use those only locally in function (or nested function calls). It's perfectly ok to pass them as references. And if you need to save them for future-use (In returned `Reload`) you clone value and store it.

Reply:
Please point me to the exact location where I could just pass a reference. There is exactly one point, `SimpleFormat::import`, otherwise I can't.

Reply:
`name` and `options` in `Format::import` and `options` in `SimpleFormat`.
And then in `impl Reload for SingleFile` you can reuse `SingleFile` value instead of creating new one

Reply:
As I've explained, you can't do that.